From 58ff970c8d0f86cfd3a62a4910d388fc388f0ab9 Mon Sep 17 00:00:00 2001
From: Robin Gong <B38343@freescale.com>
Date: Thu, 28 Jun 2012 14:35:37 +0800
Subject: [PATCH 1656/2463] ENGR00215188-4 PFUZE : remove related old code in
 pfuze driver

1.remove related old code in pfuze driver
2.add i2c write retry as i2c read retry.
Signed-off-by: Robin Gong <B38343@freescale.com>
---
 drivers/mfd/pfuze-core.c               |   10 +++++++---
 drivers/regulator/pfuze100-regulator.c |   19 -------------------
 2 files changed, 7 insertions(+), 22 deletions(-)

diff --git a/drivers/mfd/pfuze-core.c b/drivers/mfd/pfuze-core.c
index 9c7a988..febd4b0 100644
--- a/drivers/mfd/pfuze-core.c
+++ b/drivers/mfd/pfuze-core.c
@@ -109,12 +109,16 @@ int pfuze_reg_write(struct mc_pfuze *mc_pfuze, unsigned int offset,
 		    unsigned char val)
 {
 	unsigned char buf[2];
-	int ret;
+	int ret, i;
 
 	buf[0] = (unsigned char)offset;
 	memcpy(&buf[1], &val, 1);
-
-	ret = i2c_master_send(mc_pfuze->i2c_client, buf, 2);
+	for (i = 0; i < PFUZE_I2C_RETRY_TIMES; i++) {
+		ret = i2c_master_send(mc_pfuze->i2c_client, buf, 2);
+		if (ret == 2)
+			break;
+		msleep(1);
+	}
 	if (ret != 2) {
 		dev_err(&mc_pfuze->i2c_client->dev, "write failed!:%i\n", ret);
 		return ret;
diff --git a/drivers/regulator/pfuze100-regulator.c b/drivers/regulator/pfuze100-regulator.c
index e2aaa2d..34ca0fb4 100644
--- a/drivers/regulator/pfuze100-regulator.c
+++ b/drivers/regulator/pfuze100-regulator.c
@@ -657,21 +657,6 @@ static int __devexit pfuze100_regulator_remove(struct platform_device *pdev)
 	return 0;
 }
 
-#ifdef CONFIG_MX6_INTER_LDO_BYPASS
-int cpu_freq_suspend_in;	/*judge for in suspend or not*/
-static int pfuze100_regulator_suspend(struct platform_device *pdev,
-					pm_message_t state)
-{
-	cpu_freq_suspend_in = 1;
-	return 0;
-}
-
-static int pfuze100_regulator_resume(struct platform_device *pdev)
-{
-	cpu_freq_suspend_in = 0;
-	return 0;
-}
-#endif
 static struct platform_driver pfuze100_regulator_driver = {
 	.driver = {
 		   .name = "pfuze100-regulator",
@@ -679,10 +664,6 @@ static struct platform_driver pfuze100_regulator_driver = {
 		   },
 	.remove = __devexit_p(pfuze100_regulator_remove),
 	.probe = pfuze100_regulator_probe,
-#ifdef CONFIG_MX6_INTER_LDO_BYPASS
-	.suspend = pfuze100_regulator_suspend,
-	.resume = pfuze100_regulator_resume,
-#endif
 };
 
 static int __init pfuze100_regulator_init(void)
-- 
1.7.7.4

